chemprop/features/features_generators.py

```diff
@@ -50,84 +50,131 @@ def get_available_features_generators() -> List[str]:
 
 
 @register_features_generator('morgan')
-def morgan_binary_features_generator(mol: Molecule,
+def morgan_binary_features_generator(mol_data: Union[Molecule, List[List[Molecule]]],
                                      radius: int = MORGAN_RADIUS,
                                      num_bits: int = MORGAN_NUM_BITS) -> np.ndarray:
     """
     Generates a binary Morgan fingerprint for a molecule.
-
-    :param mol: A molecule (i.e., either a SMILES or an RDKit molecule).
+    :param mol_data: A molecule (i.e., either a SMILES or an RDKit molecule).
     :param radius: Morgan fingerprint radius.
     :param num_bits: Number of bits in Morgan fingerprint.
     :return: A 1D numpy array containing the binary Morgan fingerprint.
     """
-    mol = Chem.MolFromSmiles(mol) if type(mol) == str else mol
-    features_vec = AllChem.GetMorganFingerprintAsBitVect(mol, radius, nBits=num_bits)
-    features = np.zeros((1,))
-    DataStructs.ConvertToNumpyArray(features_vec, features)
+
+    def fingerprint_single_molecule(m: Molecule) -> np.ndarray:
+        m = Chem.MolFromSmiles(m) if type(m) == str else m
```

Comment:
```suggestion
        m = Chem.MolFromSmiles(m) if isinstance(m, str) else m
```
this applies to the rest of the functions as well